tests/Feature/UserTest.php

```diff
@@ -0,0 +1,113 @@
+            'dietary_restrictions' => null,
+            'school' => null,
+            'major' => null,
+            'study_level' => null,
+            'special_needs' => null,
+            'bio' => null,
+        ]);
+    }
+
+    public function test_user_can_edit_his_profile()
+    {
+        // Prepare
+        $requestData = [
+            'phone_number' => '915524851',
+            'gender' => 'm',
+            'birthdate' => Carbon::now()->subYears(20)->toDateString(),
+            'dietary_restrictions' => 'Don\'t have any dietary restrictions',
+            'school' => 'University of Minho',
+            'major' => 'Informatics',
+            'study_level' => 'MsC',
+            'special_needs' => 'Only love',
+            'bio' => 'A very specific description',
+        ];
+
+        // Execute
+        $response = $this->actingAs($this->user)
+            ->put(route('users.update'), $requestData);
+
+        // Assert
+        $response->assertRedirect();
+        $this->assertFalse(app('session.store')->has('errors'));
+        $this->assertUserWasUpdated($requestData);
+    }
+
+    public function test_an_under_age_user_can_not_edit_his_profile()
+    {
+        // Prepare
+        $requestData = [
+            'phone_number' => '915524851',
+            'gender' => 'm',
+            'birthdate' => Carbon::now()->subYears(15)->toDateString(),
+            'dietary_restrictions' => 'Don\'t have any dietary restrictions',
+            'school' => 'University of Minho',
+            'major' => 'Informatics',
+            'study_level' => 'MsC',
+            'special_needs' => 'Only love',
+            'bio' => 'A very specific description',
+        ];
+
+        // Execute
+        $response = $this->actingAs($this->user)
+            ->put(route('users.update'), $requestData);
+
+        // Assert
+        $response->assertRedirect();
+        $this->assertUserWasNotUpdated($requestData);
+        $response->assertSessionHasErrors('birthdate');
+    }
+
+    /*
+     * Assert user model was updated.
+     */
+    protected function assertUserWasUpdated(array $data)
+    {
+        $this->user->refresh();
+        $this->assertEquals($data['phone_number'], $this->user->phone_number);
+        $this->assertEquals($data['gender'], $this->user->gender);
+        $this->assertEquals($data['birthdate'], $this->user->birthdate);
+        $this->assertEquals($data['dietary_restrictions'], $this->user->dietary_restrictions);
+        $this->assertEquals($data['school'], $this->user->school);
+        $this->assertEquals($data['major'], $this->user->major);
+        $this->assertEquals($data['study_level'], $this->user->study_level);
+        $this->assertEquals($data['special_needs'], $this->user->special_needs);
+        $this->assertEquals($data['bio'], $this->user->bio);
+    }
+
+    /*
+     * Assert user model was not updated.
+     */
+    protected function assertUserWasNotUpdated(array $data)
```

Comment:
This would be stronger if you get the user from the database, using `fresh()` and compare that it matches the old user values.